pkg/apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -394,6 +394,7 @@ type EKSStatus struct {
 	PrivateRequiresTunnel         *bool                       `json:"privateRequiresTunnel"`
 	ManagedLaunchTemplateID       string                      `json:"managedLaunchTemplateID"`
 	ManagedLaunchTemplateVersions map[string]string           `json:"managedLaunchTemplateVersions"`
+	GeneratedNodeRole             string                      `json:"generatedNodeRole"`
```

Comment:
Why do we need to add generated noderole to both the cluster status and the EKSClusterConfig object in the operator? If rancher is generating the role then I assume the EKSCC object shouldn't need it and rancher will just write it to NodeRole? I see we do this in other places but do not remember what the purpose was.

Reply:
The original idea here was to design the `GeneratedNodeRole` field the way `ManagedTemplateID` is. The EKSCC object needs the generated node role because it checks for the generated value on `config.Status.GeneratedNodeRole.` If the user didn't specify a custom node role for a node group in the rancher ui, and a generated node role has already been created (which the EKS operator would only know by looking at the EKSCC status), then it creates the node group with that generated role.

Reply:
Oh, I forgot that on the cluster obj we only have the spec and not the status from the ekscc object. Sounds good!